geopandas/plotting.py

```diff
@@ -583,15 +583,14 @@ def plot_dataframe(
         categorical = True
         binedges = [values[~nan_idx].min()] + binning.bins.tolist()
         if legend_kwds is not None and 'labels' in legend_kwds:
-            if len(legend_kwds['labels']) != classification_kwds['k']:
-                raise AttributeError('Number of labels must match number of bins, '
-                                     'received {} labels for {} bins'.
-                                     format(len(legend_kwds['labels']),
-                                            classification_kwds['k']))
+            if len(legend_kwds['labels']) != binning.k:
+                raise ValueError('Number of labels must match number of bins, '
+                                 'received {} labels for {} bins'.
+                                 format(len(legend_kwds['labels']),
+                                        binning.k))
             else:
                 categories = list(legend_kwds['labels'])
-                # The 'label' keyword raises an error
-                del legend_kwds['labels']
+                legend_kwds.pop('labels')
```

Comment:
I meant to do `categories = legend_kwds.pop('labels')`.

Reply:
Ha OK I didn't get it, it's indeed easier

Reply:
Maybe `categories = list(legend_kwds.pop('labels'))` ? If you want to make, e.g. `labels=range(k)` this will raise en Error if you also have to handle NaN values because of [this](https://github.com/geopandas/geopandas/blob/0a1799d986861e55c26e38b6273bfb70012c7d1e/geopandas/plotting.py#L705)

Reply:
Oh, yes. Sure.